base.py

```diff
@@ -992,25 +992,30 @@ def update_webgpu_cts_expectations(project_root_dir, gpu_device_id):
             Util.warning(f'{expectation_file} does not exist')
             return
 
-        tag_header_scope = False
+        tag_header_scope = True
         has_new_gpu_tag = False
+        update_flag = False
+        update_comment = f'LOCAL UPDATE FOR INTEL GEN{generation} GPUS'
         for line in fileinput.input(expectation_file, inplace=True):
-            # Skip the update if the new gpu tag already exists
-            if has_new_gpu_tag:
+            # Skip if the expectation file had been updated.
+            if update_flag:
                 sys.stdout.write(line)
                 continue
 
-            if re.search(new_gpu_tag, line):
-                has_new_gpu_tag = True
-            elif re.search('BEGIN TAG HEADER', line):
-                tag_header_scope = True
-            elif re.search('END TAG HEADER', line):
-                tag_header_scope = False
-            elif re.search(old_gpu_tag, line):
-                if tag_header_scope:
+            if tag_header_scope:
+                if re.search(new_gpu_tag, line):
+                    has_new_gpu_tag = True
```

Comment:
We can't handle if new_gpu_tag is in next line of old_gpu_tag based on current impl. So we can remove has_new_gpu_tag and just put "pass" here.

Reply:
It's difficult to handle the new gpu tag in next line because we open the file using inplace mode, but it does not matter if there are duplicate gpu tags in the header.